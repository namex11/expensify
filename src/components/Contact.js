import React from "react";

const Contact = ()=>(
    <div>
        <p>This is my contacts</p>
    </div>
)

export default Contact;