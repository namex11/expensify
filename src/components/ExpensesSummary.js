import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import '../locales/lt';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

const ExpensesSummary = (props) => {
    const expenseWord = props.expensesCount <= 1 ? 'expense' : 'expenses'
    return (
        <div>
            <p>{`Viewing ${props.expensesCount} ${expenseWord} totaling ${props.expensesTotal}`}</p>
        </div>
    )
}


const mapStateToProps = ((state) => {
    return {
        expensesCount: selectExpenses(state.expenses, state.filters).length,
        expensesTotal: numeral(selectExpensesTotal(selectExpenses(state.expenses, state.filters)) / 100).format('0,0[.]00 $')
    }
})

export default connect(mapStateToProps)(ExpensesSummary)