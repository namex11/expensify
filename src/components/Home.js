import React from "react";

const Home =()=>(
    <div>
        <p>This is my home page</p>
    </div>
)

export default Home;