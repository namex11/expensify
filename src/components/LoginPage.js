import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) =>(
    <div className='box-layout'>
        <button onClick={startLogin}>Login with Google account</button>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)