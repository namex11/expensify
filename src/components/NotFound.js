import React from "react";
import { Link } from "react-router-dom";

const NotFound = ()=>(
    <div>
        <p>404- page not found</p>
        <Link to="/" exact="true"> Go home bitch </Link>
    </div>
)

export default NotFound;