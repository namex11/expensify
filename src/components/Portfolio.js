import React from "react";
import { Link } from "react-router-dom";

const Portfolio = ()=>(
    <div>
        <p>This is my Portfolio</p>
        <Link to="/portfolio/1" > To my first </Link>
        <Link to="/portfolio/2"> To my second </Link>
    </div>
)




export default Portfolio;