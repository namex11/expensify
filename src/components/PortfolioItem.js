import React from "react";


const PortfolioItem = (props)=>(
    <div>
        This is my portfolio nr {props.match.params.id}
    </div>
)

export default PortfolioItem;