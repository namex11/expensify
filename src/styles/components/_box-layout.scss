.box-layout {
    background: url('/images/background.jpg');
    height: 100vh;
    width: 100vw;
}