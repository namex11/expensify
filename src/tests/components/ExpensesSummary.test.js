import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import { shallow } from 'enzyme';

test("should render ExpensesSummary with 1 expense", () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={1256} />);
    expect(wrapper).toMatchSnapshot();
});

test("should render ExpensesSummary with multiple expenses", ()=>{
    const wrapper =shallow(<ExpensesSummary expensesCount={24} expensesTotal={456945228} />);
    expect(wrapper).toMatchSnapshot();
});