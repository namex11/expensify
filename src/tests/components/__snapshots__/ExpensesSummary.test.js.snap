// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    12,56 €
  </h1>
</div>
`;

exports[`should render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    24
     
    expenses
     totaling 
    4 569 452,28 €
  </h1>
</div>
`;
